### ISSUE TITLE
0.6.22 PNG file - white / transparent background still burned as black 

### ISSUE THREAD

**Issue by Sophist-UK** [meerk40t/meerk40t#416]
[Circular keyfob dragon.zip](https://github.com/meerk40t/meerk40t/files/6474003/Circular.keyfob.dragon.zip)

The tail image burns fine, the head image background is white and burns as black. Previous version had transparent background to head and it burned as black also.

**Comment by tatarize** [comment]
Hm. 0.6.22 should have fixed issue #390 which tended to have some minor issues there. There also might be an OS component to the problem since sometimes you get different backgrounds based in part on what color a particular implementation calls default.

**Comment by tatarize** [comment]
Are the images being classed as rasters for you and combined? Or is it kicking in when classed as images? The head seems to be rastering okay in image. Unless it's somehow doing the proper smaller region but then coloring the transparency as black within the smaller region.

**Comment by tatarize** [comment]
Also, if it's in a raster, can you try right clicking and go to the bottom with Make Raster Image and see if it produces an image with black there for you. I can't reproduce this failing. Also, what OS since that sometimes make a difference. I think Linux had some regions getting blacks where they shouldn't have had them.

**Comment by Sophist-UK** [comment]
Sorry - it appears that the background was always transparent - and I mistakenly had not changed it to white.

O/S is Windows 10 1909.

Before #355 I used to get black background burns on every transparent PNG background. After #355 it worked fine until today and this specific image. I cannot find a menu item in 0.6.22 for Make Raster Image.

**Comment by Sophist-UK** [comment]
Here is the dragon head with transparent bg.
[Circular keyfob dragon bee.zip](https://github.com/meerk40t/meerk40t/files/6474354/Circular.keyfob.dragon.bee.zip)

**Comment by tatarize** [comment]
![make-raster](https://user-images.githubusercontent.com/3302478/118223766-dcc40680-b436-11eb-9692-332a7436967e.png)

![make-raster2](https://user-images.githubusercontent.com/3302478/118224037-34fb0880-b437-11eb-82f8-a249b833e735.png)

Make raster still went to white. So I'm thinking if it's a thing, that it'll be in the actual images themselves. The rastering routines have methods for different image types, so 1-bit black and white works like on and off, and grayscale L works like PPI applied to a shade. RGB should convert to grayscale sort of and work a bit like that. RGBA might have an issue within that. But, I'm pretty sure it should be RGB with the A value being used to attenuate the values.

`return (1.0 - (pixel[0] + pixel[1] + pixel[2]) / 765.0) * pixel[3] / 255.0` where 0, 1, and 2 are RGB and 3 is A. That seems like it should be correct. A value of 0.

Well, it's certainly something. Tested and it clearly is cutting that as black.

**Comment by tatarize** [comment]
Okay, the dragon head image in a mode P image which gets:

```python
                        v = p[pixel * 3] + p[pixel * 3 + 1] + p[pixel * 3 + 2]
                        return 1.0 - v / 765.0
```

As the mode, but that discounts the chance of there being an alpha layer, but there clearly is one. Which identifies the bug. I'll lookup how to deal with that type of mode, when it has alpha.

**Comment by tatarize** [comment]
I made the seemingly required update but it's weird and I'm going to have to debug it harder to figure out what's wrong. It weirdly goes over to the right hand side seemingly for no reason. But, if I did "image remove blue" on the image (the image has no blue and should be fine) it somehow rastered fine since it converted to RGBA. I still have something to get to the bottom of there, but it's closer.

**Comment by diamened** [comment]
I'm seeing the same problem here. The attached image classifies as 'image' and has a PNG with transparent background in it. But the laser engraves the transparent areas black. The white parts are ok.

[Viking Runestone 02.svg.zip](https://github.com/meerk40t/meerk40t/files/7147129/Viking.Runestone.02.svg.zip)

**Comment by tatarize** [comment]
Checked in the 7.1 version and this simulates those areas as black, implying it would absolutely want to raster the edges there with a hard laser-on.

**Comment by tatarize** [comment]
The 0.6.23 version appears to have treated the image correctly though that version is still beta, and where this issue was expected to be fixed.

**Comment by tatarize** [comment]
0.6.22 looked to do that correctly. It, however, did not.

![0 6 22](https://user-images.githubusercontent.com/3302478/132937373-a71e3adb-7ff4-40db-b587-730fed5c000f.png)

Testing. 0.6.23 fixes this issue. 
![0 6 23](https://user-images.githubusercontent.com/3302478/132937352-66497baa-1cff-4912-a341-5e145798e37b.png)

Should push up the publication and test the rest of the code. And test that this is not an issue in 0.7.x

**Comment by tatarize** [comment]
I tested and published the old 0.6.23 code that fixed this.

https://github.com/meerk40t/meerk40t/releases/tag/0.6.23

Please confirm that corrects it (It does). And I'll go ahead and close this issue.

**Comment by diamened** [comment]
The 0.6.23 version works perfectly. Thank you
